modules/core/src/umatrix.cpp

```diff
@@ -1212,6 +1218,9 @@ void UMat::copyTo(OutputArray _dst, InputArray _mask) const
                              ocl::memopTypeToStr(depth()), cn, mcn,
                              haveDstUninit ? " -D HAVE_DST_UNINIT" : "");
 
+        std::shared_ptr<ocl::OpenCLExecutionContext> pExecCtx = std::static_pointer_cast<ocl::OpenCLExecutionContext>(u->allocatorContext);
+        ocl::OpenCLExecutionContextScope scope(pExecCtx ? *pExecCtx.get() : ocl::OpenCLExecutionContext::getCurrent());
```

Comment:
> OpenCLExecutionContextScope

We should not change execution scope here.

If `src`'s scope is not the same (as dst/thread), then `.clone()` src UMat first.
If `mask`'s scope is not the same (as as dst/thread), then `.clone()` mask UMat first.

Reply:
i see!

Reply:
Done, and i used the same solution for convertTo